pkg/checks/checks.go

```diff
@@ -0,0 +1,106 @@
+package checks
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	coreclient "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// WaitForHTTPStatusOk poll until URL returns 200
+func WaitForHTTPStatusOk(ctx context.Context, transport http.RoundTripper, urltocheck string) error {
+	cli := &http.Client{
+		Transport: transport,
+		Timeout:   10 * time.Second,
+	}
+
+	req, err := http.NewRequest("GET", urltocheck, nil)
+	if err != nil {
+		return err
+	}
+
+	for {
+		resp, err := cli.Do(req)
+		if err, ok := err.(*url.Error); ok && (err.Timeout() || err.Err == io.EOF || err.Err == io.ErrUnexpectedEOF) {
+			time.Sleep(time.Second)
+			continue
+		}
+		if err != nil {
+			return err
+		}
+		if resp.StatusCode == http.StatusOK {
+			return nil
+		}
+
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-time.After(time.Second):
+		}
+	}
+}
+
+// WaitForPodsRunning verifies all pods are in running state
+func WaitForPodsRunning(ctx context.Context, coreclient *coreclient.CoreV1Client, namespace string) error {
+
+	whitelistedNamespaces := []string{
+		"openshift-",
+		"kube-",
+		"default",
+	}
+
+	pods, err := coreclient.Pods(namespace).List(metav1.ListOptions{})
```

Comment:
we didn't get a chance to catch up today, sorry. I think you had some questions about this.  I'm expecting to see you looping through statefulsets, daemonsets, etc. using the apps client and not the core client.
you could use the clientset at https://github.com/jim-minter/azure-helm/blob/master/vendor/k8s.io/client-go/kubernetes/clientset.go#L399, that gives you nearly every k8s client there is.  Good to help developing.
For the final thing you're better off using more scoped clients (easier for testing).  In this case, https://github.com/jim-minter/azure-helm/blob/master/vendor/k8s.io/client-go/kubernetes/typed/apps/v1/apps_client.go#L63

Reply:
I was able to find the statefulsets and daemonsets by using the apps_client.go you mentioned.  

My question would be, what is the advantage of getting those objects from the apps_client when the coreclient fetches the pods which are from those objects?  The owner reference on the pod has a `kind` field which tells you about the parent object.  

Is it specifically that daemonsets and statefulsets don't deploy when there are no matching selectors?  If this is the case, I'd lean to verifying there are nodes labeled appropriately so that the selectors are correct.

I am fine proceeding but would like your insight as to which is the preferred so that I understand.

Thanks @jim-minter

Reply:
It's a mess and the answer is almost certainly that both clients are needed. I think it's probably best to iterate the daemonsets/deployments/etc and then resort to checking the pods only if necessary because the info isn't in the parent's status, but there are probably multiple ways to skin this particular cat.